modules/mpi/tutorials/ospMPIDistribTutorialVolumeHDF5.cpp

```diff
@@ -90,13 +90,61 @@ int main(int argc, char **argv)
         },
         nullptr);
 
+    std::cerr << "ABOUT TO READ" << std::endl;
+
     // open the hdf5 file, read the data and create the ospray Volume object
     ChomboHDF5::Reader hdf5reader(argv[1], mpiRank, mpiWorldSize, argv[2]);
+    std::cerr << "READ!" << std::endl;
+
+    VolumeBrick brick;
+
+    //convert from int to float as required by ospray
+    //get the world bounds while we are at it
+    std::vector<box3f> myregions;
+    vec3f min{MAXFLOAT};
+    vec3f max{-MAXFLOAT};
+    for (auto x : hdf5reader.m_blockBounds) {
+        std::cerr << x.lower << " " << x.upper << std::endl;
+        if (x.lower.x < min.x)
+            min.x = x.lower.x;
+        if (x.lower.y < min.y)
+            min.y = x.lower.y;
+        if (x.lower.z < min.z)
+            min.z = x.lower.z;
+        if (x.lower.x > max.x)
+            max.x = x.lower.x;
+        if (x.lower.y > max.y)
+            max.y = x.lower.y;
+        if (x.lower.z > max.z)
+            max.z = x.lower.z;
+        myregions.push_back(box3f(x.lower, x.upper));
+    }
+    //std::cerr << min << " " << max << std::endl;
+
+    brick.brick = hdf5reader.getVolume();
+    brick.bounds = myregions;
+
+    worldBounds = box3f(min*0.25, max*0.25); //seems like we are missing the block scale somewhere
```

Comment:
For some reason I thought this could be easily deduced from what is read in `ChomboHDF5::Reader` but now the only way I can think of doing so is from iterating through all the boxes on the coarsest level and finding the one with the largest upper corner which is a bit of a faff. This information is stored in the HDF5 file so I will add some code to read it in.

Reply:
Ok, I've now added a function `ChomboHDF5::Reader::getDomainBounds` which should return a suitable `box3f` you can set the `worldBounds` to.

Reply:
updated, ty.